### ISSUE TITLE
standardize data-at-shortcutkeys ??

### ISSUE THREAD

**Issue by jnurthen** [w3c/aria#1257]
I recently heard about this
https://support.freedomscientific.com/Training/Surfs-Up/BypassNavigationQuickKeys.htm

Is this the same thing covered by aria-keyshortcuts (I think not but haven't looked into it in detail).
As this was apparently done for facebook maybe @mcking65 can shed some more light on it.

**Comment by carmacleod** [comment]
They sure do seem similar, although the JAWS `data-at-shortcutkeys` feature does not seem to support modifier keys (i.e. it seems to only support single-character Navigation Quick Keys).
(Heh, I'll bet the corresponding `SHIFT+key` still does reverse navigation - just guessing, though).

Not sure if JAWS supports [aria-keyshortcuts](https://w3c.github.io/aria/#aria-keyshortcuts) at all? This [JAWS ARIA Support table](https://freedomscientific.github.io/VFO-standards-support/aria.html) (last updated a year ago) has `?` under `aria-keyshortcuts`, and this [Accessibility Support site](https://a11ysupport.io/) says "We do not currently have any data on this feature.".

TBH, I don't know if any AT supports `aria-keyshortcuts`?

**Comment by JAWS-test** [comment]
> Not sure if JAWS supports aria-keyshortcuts at all? 

JAWS supports aria-keyshortcuts very well. Provided the property is correctly passed from the browser to the Accessibility API, JAWS outputs the accesskeys during tab navigation, when reading with the virtual cursor, and sometimes even in the element overview. The output works for interactive elements as well as for non-interactive elements (such as span). 
Unfortunately, aria-keyshortcuts is not supported by Firefox and IE 11. But this is a browser bug. JAWS only outputs access keys correctly with Chrome and Edge (Chromium)

**Comment by JAWS-test** [comment]
I think the data-at-shortcutkeys deal between Freedom Scientific and Facebook is very problematic because
* the technique will not work with other screen readers
* a new attribute has been introduced that is not in line with the ARIA standard
* Many JAWS users will not know the feature and will not understand why their JAWS keys no longer work as usual
* There is no need for the attribute because there are enough well-known methods in JAWS for passing keys to the browser
* it is very questionable whether the Facebook implementation in connection with data-at-shortcutkeys takes into account the requirements of [WCAG 2.1.4]( https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html). Since I don't have a Facebook account, I can't check this myself

**Comment by carmacleod** [comment]
Thank-you @JAWS-test for the useful info. Good to know that JAWS works well with `aria-keyshortcuts` (in supporting browsers), and thanks for pointing out how these are 2 very different things.

> I think the data-at-shortcutkeys deal between Freedom Scientific and Facebook is very problematic

You are not the only one who thinks this is problematic. @mcking65 agrees with you, as do others in the ARIA Working Group.

**Comment by carmacleod** [comment]
@JAWS-test You said:

> However, data-at-shortcutkeys does not work in my test environment (JAWS 2019 and JAWS 2020 with IE, Firefox, Edge, Chrome).

Do you think the feature may have been removed? i.e. did your test case work with previous versions of JAWS?

**Comment by JAWS-test** [comment]
> Do you think the feature may have been removed?

I don't think so, because the JAWS setting to enable this feature is still available: "Allow Web Application Reserved Keystrokes" (see https://support.freedomscientific.com/Training/Surfs-Up/BypassNavigationQuickKeys.htm)

**Comment by carmacleod** [comment]
>  the JAWS setting to enable this feature is still available

Good point. If the feature was removed, the setting would also be removed.

**Comment by carmacleod** [comment]
@jamesn Can this be closed? It does not seem generally useful.

**Comment by mcking65** [comment]
Just for the record, I support closing this as I do not support standardizing the data attribute for shortcut keys.

It was a tactical solution to a practical problem negotiated between Facebook and Freedom Scientific prior to when I became a Facebook employee.

I recently started working with Vispero to phase out use of this feature on Facebook.

As far as I know, the only other site using it is Twitter.
